app/src/main/java/org/dicio/dicio_android/skills/navigate/NavigateProcessor.java

```diff
@@ -0,0 +1,79 @@
+package org.dicio.dicio_android.skills.navigate;
+
+import static org.dicio.dicio_android.Sentences_en.navigate;
+
+import org.dicio.dicio_android.util.StringUtils;
+import org.dicio.skill.chain.IntermediateProcessor;
+import org.dicio.skill.standard.StandardResult;
+
+import java.util.List;
+
+public class NavigateProcessor extends IntermediateProcessor<StandardResult, NavigateOutput.Data> {
+    @Override
+    public NavigateOutput.Data process(final StandardResult data) throws Exception {
+        final NavigateOutput.Data result = new NavigateOutput.Data();
+
+        if (data == null) {
+            result.failed = true;
+            return result;
+        }
+
+        String placeToNavigate = data.getCapturingGroup(navigate.where);
+
+        if (placeToNavigate == null) {
+            result.failed = true;
+            return result;
+        }
+
+        placeToNavigate = placeToNavigate.trim();
+
+        if (StringUtils.isNullOrEmpty(placeToNavigate)) {
+            result.failed = true;
+            return result;
+        }
+
+        final List<Object> textWithNumbers = ctx().requireNumberParserFormatter()
+                .extractNumbers(data.getCapturingGroup("where")).get();
+
+        // It seems that we detect each digit of the numerical portion of an address as a separate
+        // string, resulting in "9 5 4 6 1 9 avenue" when the speech was actually "9546 19 avenue".
+        // Here we will remove the spaces from between the numbers, so we will end up with
+        // "954619 avenue". This is not ideal, but it seems slightly better than before.
+        // Can we detect addresses better?
+        placeToNavigate = "";
+        int i = 0;
+        while (i < textWithNumbers.size()) {
+            final Object currentItem = textWithNumbers.get(i);
+            if (!currentItem.toString().equals(" ")) {
+                if (i == 0 || isInteger(currentItem.toString())) {
+                    placeToNavigate += currentItem.toString().trim();
```

Comment:
I don't think you need `isInteger` and `toString` here. `extractNumbers` already takes care of extracting numbers (class `org.dicio.numbers.util.Number`) and separating them from `String`s (the usual `java.lang.String`). You may want to take a look at `CalculatorProcessor` for reference. But I guess `dicio-numbers` would benefit from a bit more documentation ;-)

Reply:
With the space removal code (that I described in a separate comment) I do need them, because I need to know for each string if it contains a number or not, so I can decide whether or not to remove spaces between it and adjacent numbers.

In my other comment I proposed removing the space-removal code since it isn't perfect anyway. And if I do that, then I'll be removing this as well.

Reply:
This has been resolved due to the other changes that I mentioned.